src/core_ocean/analysis_members/mpas_ocn_lagrangian_particle_tracking.F

```diff
@@ -705,44 +712,73 @@ subroutine ocn_compute_lagrangian_particle_tracking(domain, timeLevel, err)!{{{
             ! interpolate the scalars now (assumes that scalar value is constant within a particular cell)
-            call interp_cell_scalars(iLevelBuoyancy, maxLevelCell(iCell), buoyancyInterp, buoyancyTimeInterp(:,iCell), &
+            call interp_cell_scalars(iLevel, maxLevelCell(iCell), buoyancyInterp, buoyancyTimeInterp(:,iCell), &
               zMid(:,iCell), zLevelParticle)
             !}}}
           else
             ! make sure final zLevelParticle is ok so that it can't extent past zMid range
 #ifdef MPAS_DEBUG
             call mpas_timer_start("mpas_get_vertical_idLPT")
 #endif
-            iLevel = mpas_get_vertical_id(maxLevelCell(iCell), zLevelParticle, zMid(:,iCell))
+            if (verticalTreatment == 3) then !('passiveFloat')
+              ! constrain vertical level to be within dz/2 of ssh or bottom depth
+              zLevelParticle = min(max(zMid(maxLevelCell(iCell),iCell), zLevelParticle), zMid(1,iCell))
+            end if
+            iLevel = mpas_get_vertical_id(maxLevelCell(iCell), -zLevelParticle, -zMid(:,iCell))
 #ifdef MPAS_DEBUG
             call mpas_timer_stop("mpas_get_vertical_idLPT")
 #endif
           end if
 
-          ! compute necessary information for autocorrelation !{{{
-          ! get the updated velocity
-          ! ensure that buoyancy is fixed for each run
-          buoyancyInterp = buoyancyParticle
-          ! we just need the last part of the velocity field interpolation because we are at the end of the timestep
-          timeInterpOrder = 2
-          timeCoeff(1) = 0.0_RKIND
-          timeCoeff(2) = 1.0_RKIND
-          ! return interpolated horizontal velocity "particleVelocity" and vertical velocity "particleVelocityVert"
-          ! noting we use the final positions
 #ifdef MPAS_DEBUG
-          call mpas_timer_start("velocity_time_interpolationLPT")
+          call mpas_timer_start("verifyParticleDepthInDomainLPT")
 #endif
-          call velocity_time_interpolation(particleVelocity, particleVelocityVert, &
-            diagnosticsPool, lagrPartTrackFieldsPool, &
-            timeInterpOrder, timeCoeff, iCell, iLevel, buoyancyInterp, maxLevelCell, &
-            verticalTreatment, indexLevel, nCellVertices, verticesOnCell, boundaryVertex, &
-            particlePosition, zLevelParticle, zMid, zTop, vertVelocityTop, xVertex, yVertex, zVertex, &
-            meshPool, areaBArray)
+          !verify particles are still in domain !{{{
+          ! check and verify that free floating particle hasn't gone out of boundary
+          if (iLevel == -1) then
+            call mpas_log_write(' Warning: Particle $i location of $r is higher than zMid of $r', realArgs=(/ zLevelParticle, zMid(1, iCell) /), intArgs=(/iCell/))
+          end if
+          if (iLevel == 0) then
+            call mpas_log_write(' Warning: Particle $i location of $r is deeper than zMid at bottom $r', realArgs=(/ zLevelParticle, zMid(maxLevelCell(iCell),iCell) /), intArgs=(/iCell/))
+          end if
```

Comment:
@bradyrx, this is an example of code that probably needs commented out (above) as we teach you about PRs.

Reply:
It is now wrapped in a debug statement, so this should be a non-issue.